### PR TITLE
cmd/evm: make evm blocktest output logs if so instructed

### PR DESCRIPTION
**Pull request by holiman**
This PR fixes https://github.com/ethereum/go-ethereum/issues/27363, and spits out logs while the evm is executing

### CODE REVIEW

File: tests/block_test.go [hunk 1]
```diff
@@ -48,10 +48,10 @@ func TestBlockchain(t *testing.T) {
 	bt.skipLoad(`.*randomStatetest94.json.*`)
 
 	bt.walk(t, blockTestDir, func(t *testing.T, name string, test *BlockTest) {
-		if err := bt.checkFailure(t, test.Run(false)); err != nil {
+		if err := bt.checkFailure(t, test.Run(false), nil); err != nil {
```
Comment (MariusVanDerWijden) [inline tests/block_test.go:51]:
```suggestion
		if err := bt.checkFailure(t, test.Run(false, nil)); err != nil {
```

File: tests/block_test.go [hunk 2]
```diff
@@ -48,10 +48,10 @@ func TestBlockchain(t *testing.T) {
...
 			t.Errorf("test without snapshotter failed: %v", err)
 		}
-		if err := bt.checkFailure(t, test.Run(true)); err != nil {
+		if err := bt.checkFailure(t, test.Run(true), nil); err != nil {
```
Comment (MariusVanDerWijden) [inline tests/block_test.go:54]:
```suggestion
		if err := bt.checkFailure(t, test.Run(true, nil)); err != nil {
```

**Review by MariusVanDerWijden — COMMENTED** [review]
LGTM, (ci is read because of two typos)

**Review by MariusVanDerWijden — APPROVED** [review]
LGTM, (ci is read because of two typos)

**Comment by holiman** [comment]
@d-xo works for you?

**Pull request merged**